apps/web/pages/organizations/[organizationId]/invoices/[invoiceId]/components/invoiceItemList.tsx

```diff
@@ -141,122 +141,131 @@ export const InvoiceItemList = ({ invoice }: InvoiceItemListProps): JSX.Element
 
   return (
     <>
-      <table className="table text-right">
+      <table className="table">
         <thead className="bg-neutral text-sm text-neutral-content">
-          <tr className="[&_th]:border [&_th]:border-neutral">
+          <tr className="text-right [&_td:first-child]:border-r-transparent [&_th]:border [&_th]:border-neutral">
+            <th className="w-px" />
             <th />
             <th className="w-1/12">Duration</th>
-            <th className="w-1/12">Hourly Rate, €</th>
-            <th className="w-1/12">Amount, €</th>
+            <th className="w-1/12">H. Rate (€)</th>
+            <th className="w-1/12">Amount (€)</th>
           </tr>
         </thead>
         <tbody>
           {invoiceData.invoiceItems
-            .sort((a, b) => {
-              const projectCompare = a.task.project.title.localeCompare(b.task.project.title)
-              return projectCompare === 0 ? a.task.title.localeCompare(b.task.title) : projectCompare
+            .filter((invoiceItem) => {
+              const taskStartDate = new Date(invoiceItem.task.project.startDate ?? 0)
+              const taskEndDate = invoiceItem.task.project.endDate
+                ? new Date(invoiceItem.task.project.endDate)
+                : undefined
+              const invoiceStartDate = new Date(invoiceData.invoiceWorkFrom)
+              const invoiceEndDate = new Date(invoiceData.invoiceWorkUntil)
+              return taskStartDate <= invoiceEndDate && (!taskEndDate || taskEndDate >= invoiceStartDate)
```

Comment:
Let's revert the filter, as we discussed yesterday
```suggestion
              .sort((a, b) => {
              const projectCompare = a.task.project.title.localeCompare(b.task.project.title)
              return projectCompare === 0 ? a.task.title.localeCompare(b.task.title) : projectCompare
```